proto/common.proto

```diff
@@ -6,16 +6,64 @@ message Error { string cause = 1; }
 
 message Close { string channel = 1; }
 
-message Message {
-  string channel = 1;
-  bytes data = 2;
-}
+// Central piece of data
+message DataChunk { bytes data = 1; }
 
-message Id { uint64 id = 1; }
+// This message contains data that is sent from a runner to the orchestrator
+message SendingMessage {
+  uint32 localSequenceNumber = 1;
+  string channel = 2;
+  bytes data = 3;
+}
 
-message DataChunk { bytes data = 1; }
+// This message contains data that is received by a runner from the orchestrator
+message ReceivingMessage {
+  uint32 globalSequenceNumber = 1;
+  string channel = 2;
+  bytes data = 3;
+}
 
-message StreamMessage {
-  Id id = 1;
+// StreamMessage contains the sequenceNumber used to receive the streaming data
+// This message is received by the runner from the orchestrator
+message ReceivingStreamMessage {
+  uint32 globalSequenceNumber = 1;
   string channel = 2;
 }
+
+// Control message for a streaming message, to the producer
+// When the runner receives 0 it can send the first chunk
+// After each chunk it MUST await the next sequenceNumber
+message ReceivingStreamControl { uint32 streamSequenceNumber = 2; }
+
+// Control message for a streaming message, from the consumer
+// First the consumer must identify which stream it wants to consume, with the
+// globalSequenceNumber
+// Then after handling a chunk, it must send how many chunks have been handled
+message SendingStreamControl {
+  oneof inner {
+    uint32 globalSequenceNumber = 1;
+    uint32 streamSequenceNumber = 2;
+  }
+}
+
+// The first message when a runner sends a streaming message to the orchestrator
+message StreamIdentify {
+  string channel = 1;
+  uint32 localSequenceNumber = 2;
```

Comment:
It's not a `...Message` object, but having the sequence number here first and then the channel might be more consistent.
Idem for the MessageProcessed below.
(If you would update it, don't forget the readme)